### DISCUSSION TITLE
Error on IntegrationTestsFixture  and detect host configuration

### DISCUSSION THREAD

**Discussion by exile69000** [testcontainers/testcontainers-dotnet#1101] [Q&A] [answered]
I'm running tests from CI/CD on AzureDevOps.

I have two errors : "The following constructor parameters did not have matching fixture data: IntegrationTestsFixture fixture" and "Auto discovery did not detect a Docker host configuration".

I don't know what happen, on my local computer test is ok.

![image](https://github.com/testcontainers/testcontainers-dotnet/assets/72208464/c7513727-2caf-477f-a4e8-eb4f74b739f6)

![image](https://github.com/testcontainers/testcontainers-dotnet/assets/72208464/beba801d-60ae-4ce2-b6e9-f0571fc6f76c)

![image](https://github.com/testcontainers/testcontainers-dotnet/assets/72208464/dc27f578-6e1c-406c-be41-a5a37f28e2dd)

thanks for your help :)

**Comment by HofmeisterAn** [comment]
> Docker is either not running or misconfigured.

Are you running on a private CI agent? Does the CI agent contain a Docker runtime? This error message usually indicates that there is no Docker (container) runtime available. Please make sure Docker is installed.

**Comment by exile69000** [comment]
Yes I'm running on private CI agent, I created Linux Agent, on Debian server I installed Docker and Docker Desktop: 
![image](https://github.com/testcontainers/testcontainers-dotnet/assets/72208464/d87f9e9e-434a-45af-b4b9-796482cd74bf)
My Nugets TestContainers are up to date.
This is my pipeline : 
![image](https://github.com/testcontainers/testcontainers-dotnet/assets/72208464/3bfe7941-f3f0-4118-a623-9cdd2d4e5696)

Is there a command to check if Docker is correctly installed and CI Agent use Docker runtime ?

**Reply by HofmeisterAn** [reply]
> Yes I'm running on private CI agent, I created Linux Agent, on Debian server I installed Docker and Docker Desktop:

Ok, that's good. It looks like Testcontainers does not found the daemon. Can you share the output of

    docker context ls

> Is there a command to check if Docker is correctly installed and CI Agent use Docker runtime ?

`docker run hello-world` should be fine.

**Comment by exile69000** [comment]
Output of `docker context ls` : 
![image](https://github.com/testcontainers/testcontainers-dotnet/assets/72208464/1c24b033-531e-4768-9e4f-b3554177065e)

Maybe I have to specify the DOCKER_HOST in .testcontainers.properties ?

**Reply by HofmeisterAn** [reply]
Can you please reply to a comment instead of creating a new one. Hm, this looks correct. The endpoint should be detected by our [default](https://github.com/testcontainers/testcontainers-dotnet/blob/7d6bbffbab7426706a8d38c1ee783af0d2b48726/src/Testcontainers/Builders/UnixEndpointAuthenticationProvider.cs) provider. Does the user who runs the agent have permissions to access Docker (the daemon)?

**Reply by exile69000** [reply]
How I can check if my user have permissions ? It is a Linux command ? Have you got an example to check ? Thanks

**Reply by exile69000** [reply]
I think I have not permission : 
![image](https://github.com/testcontainers/testcontainers-dotnet/assets/72208464/2ab46419-2a1c-45a7-bab5-a9c03c72248d)

**Reply by HofmeisterAn** [reply]
Yep, that looks like the issue. I do not know your environment, so I am  afraid I cannot provide much assistance here. If a Docker group exists, you might want to consider adding the agent user to that group.

**Reply by exile69000** [reply]
I added user in Docker group, restart docker engine but same issue....
If I created .testcontainers.properties in my folder user like this it's good :
`docker.host=unix:///var/run/docker.sock`

**Reply by exile69000** [reply, marked as answer]
So I found solution ! I add my user in sudo group and it's working :)

**Reply by HofmeisterAn** [reply]
Please be careful; this will grant essentially administrative privileges to your build user, not always something you want.

**Reply by exile69000** [reply]
Yes you’re right ! I need to see with my network team to resolve this issue, in any case it’s working
